pyro/distributions/avf_mvn.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+from torch.distributions import constraints
+
+from pyro.distributions.torch import MultivariateNormal
+from pyro.distributions.util import sum_leftmost
+
+
+class AVFMultivariateNormal(MultivariateNormal):
+    """Multivariate normal (Gaussian) distribution with transport equation inspired control
+    variates (adaptive velocity fields).
+
+    A distribution over vectors in which all the elements have a joint Gaussian density.
+
+    :param torch.Tensor loc: D-dimensional mean vector.
+    :param torch.Tensor scale_tril: Cholesky of Covariance matrix; D x D matrix.
+    :param torch.Tensor CV: 2 x L x D tensor that parameterizes the control variate; L is an arbitrary positive integer.
+    This parameter needs to be learned (i.e. adapted) to achieve lower variance gradients.
+
+    Example usage::
+
+    CV = torch.tensor(0.1 * torch.ones(2, 1, D), requires_grad=True)
+    opt_cv = pyro.optim.Adam([CV], lr=0.1, betas=(0.5, 0.999))
+
+    for _ in range(1000):
+        d = AVFMultivariateNormal(loc, scale_tril, CV)
+        z = d.rsample()
+        cost = torch.pow(z, 2.0).sum()
+        cost.backward()
```

Comment:
this looks runnable do you want to add it to doctests? via `>>>`

Reply:
nope purposely didn't do that